core/trino-spi/src/main/java/io/trino/spi/predicate/SortedRangeSet.java

```diff
@@ -919,7 +919,9 @@ public Optional<Collection<Object>> tryExpandRanges(int valuesLimit)
         List<Range> ranges = getRanges().getOrderedRanges();
         Type type = getType();
 
-        Range typeRange = type.getRange().map(range -> Range.range(type, range.getMin(), true, range.getMax(), true)).orElse(Range.all(type));
+        Range typeRange = type.getRange()
+                .map(range -> Range.range(type, range.getMin(), true, range.getMax(), true))
+                .orElseGet(() -> Range.all(type));
```

Comment:
technically you could also cache `Range.all()` by type

Reply:
I don't know whether this would be an improvement or not. I guess it may not be.

Reply:
There are many more cases where we call `Optional.orElse` with an expensive expression, some that look really expensive. I am trying to fix them in https://github.com/trinodb/trino/pull/15301